docs/docs/examples.md

```diff
@@ -0,0 +1,308 @@
+       ```
+9. The client stores the certificate.
+
+To recap: This example shows how to
+* instantiate a signer for a hardware-backed, biometry-protected, attested key
+* instantiate a verifier
+* create, sign and verify CSRs with a custom attribute
+* extract a custom attribute from a CSR
+* create, and sign a certificate with a custom critical extension
+
+## Creating a Signed JSON Web Signature Object (`JwsSigned`)
+
+!!! info  inline end
+    This example requires the _Supreme_ KMP crypto provider and _Indispensable Josef_.
+
+In this example, we'll start with an ephemeral P-256 signer: 
+
+```kotlin
+val signer = Signer.Ephemeral {
+    ec { curve = ECCurve.SECP_256_R_1 }
+}.getOrThrow() //TODO handle error properly
+```
+
+Next up, we'll create a header and payload:
+
+```kotlin
+val header = JwsHeader(
+    algorithm = signer.signatureAlgorithm.toJwsAlgorithm().getOrThrow(),
+    jsonWebKey = signer.publicKey.toJsonWebKey()
+)
+val payload = byteArrayOf(1, 3, 3, 7)
+```
+
+Since both header and payload are fed into the signature, we need to prepare this signature input:
+
+```kotlin
+val plainSignatureInput = JwsSigned.prepareJwsSignatureInput(header, payload)
+```
+
+Now, everything is ready to be signed:
+
+```kotlin
+val signature = signer.sign(plainSignatureInput.encodeToByteArray()).signature //TODO: handle error
+JwsSigned(header, payload, signature, plainSignatureInput).serialize() // this we can verify on jwt.io 
+```
+
+As can be seen, a `JwsSigned` takes header, payload, signature, and the plain signature input as parameters.
+The reason for keeping this fourth parameter is convenience and efficiency: For one, you need this input to serialize a
+`JwsSigned`, so it would be a waste to discard it. After parsing a `JswSigned` from its serialized form, you also need the 
+`plainSignatureInput` to verify everything was signed correctly.
+
+
+## Creating a `CoseSigned` Object
+
+!!! info  inline end
+    This example requires the _Supreme_ KMP crypto provider and _Indispensable Cosef_.
+
+In this example, we'll again start with an ephemeral P-256 signer:
+
+```kotlin
+val signer = Signer.Ephemeral {
+    ec { curve = ECCurve.SECP_256_R_1 }
+}.getOrThrow() //TODO handle error properly
+```
+
+Next up, we'll create a header and payload:
+
+```kotlin
+//set KID + algorithm
+val protectedHeader = CoseHeader(
+    algorithm = signer.signatureAlgorithm.toCoseAlgorithm().getOrElse { TODO() },
+    kid = signer.publicKey.didEncoded.encodeToByteArray()
+)
+
+val payload = byteArrayOf(0xC, 0xA, 0xF, 0xE)
+```
+
+Both of these are signature inputs, so we'll construct a `CoseSignatureInput` to sign.
+
+```kotlin
+val signatureInput = CoseSignatureInput(
+    contextString = "Signature1",
+    protectedHeader = ByteStringWrapper(protectedHeader),
+    externalAad = byteArrayOf(),
+    payload = payload,
+).serialize()
+```
+
+
+Now, everything is ready to be signed:
```

Comment:
this should have taken one line

Reply:
?

Reply:
same as above; you have correctly identified that this is a really common use case, but the correct fix is to make it take one line, not to have a documentation example that takes many lines

Reply:
which is why we have an open issue for this. and since it requires a larger revamp with additional modules etc. we document it for now, because this PR is the wrong place for it